app/parse.py

```diff
@@ -8,8 +18,66 @@ class Quote:
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(asctime)s] - [%(levelname)8s]: %(message)s",
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+        logging.FileHandler("parser.log"),
+    ]
+)
+
+
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    text = quote_soup.select_one(".text").text
+    author = quote_soup.select_one(".author").text
+    tags = (
+        quote_soup.select_one(".tags").text
+        .replace(" ", "").split("\n")
+    )
+    tags = [tag for tag in tags if tag and tag != "Tags:"]
+    return Quote(text, author, tags)
+
+
+def get_num_pages(page_soup: BeautifulSoup) -> int:
+    page_soup.select_one()
+
+
+def get_all_quotes() -> list[Quote]:
+    logging.info("Getting quotes for 1 page")
+    page = requests.get(BASE_URL).content
+    first_page_soup = BeautifulSoup(page, "html.parser")
+    num_pages = 10
```

Comment:
The `num_pages` variable is hardcoded to 10. This should be dynamically determined using the `get_num_pages` function to ensure all pages are scraped.